packages/react-dom-bindings/src/events/ReactDOMEventReplaying.js

```diff
@@ -518,6 +542,20 @@ export function retryIfBlockedOn(
     const queuedTarget = queuedExplicitHydrationTargets[i];
     if (queuedTarget.blockedOn === unblocked) {
       queuedTarget.blockedOn = null;
+      const targetInst = getClosestInstanceFromNode(queuedTarget.target);
```

Comment:
This is a fairly expensive operation since it needs to backtrack potentially many nodes. If there are many queued targets, it'll also have to do that for each one each time something gets unblocked.